src/psyclone/dynamo0p3_builtins.py

```diff
@@ -367,194 +490,229 @@ def gen_code(self, parent):
 # describing these kernels is in dynamo0p3_builtins_mod.f90. This dictionary
 # can only be defined after all of the necessary 'class' statements have
 # been executed (happens when this module is imported into another).
-BUILTIN_MAP = {"axmy": DynAXMYKern,
-               "axpy": DynAXPYKern, "inc_axpy": DynIncAXPYKern,
-               "axpby": DynAXPBYKern, "inc_axpby": DynIncAXPBYKern,
-               "copy_field": DynCopyFieldKern,
-               "copy_scaled_field": DynCopyScaledFieldKern,
-               "divide_fields": DynDivideFieldsKern,
-               "inc_divide_field": DynIncDivideFieldKern,
-               "inc_field": DynIncFieldKern,
-               "inc_multiply_field": DynIncMultiplyFieldKern,
-               "inc_xpby": DynIncXPBYKern,
-               "inner_product": DynInnerProductKern,
-               "inner_self_product": DynInnerSelfProductKern,
-               "minus_fields": DynSubtractFieldsKern,
-               "multiply_fields": DynMultiplyFieldsKern,
-               "plus_fields": DynAddFieldsKern,
-               "raise_field": DynRaiseFieldKern,
-               "scale_field": DynScaleFieldKern,
-               "set_field_scalar": DynSetFieldScalarKern,
-               "sum_field": DynSumFieldKern}
+# Note: Issue #58 will introduce functionality to obtain list of supported
+# built-ins from dynamo0p3_builtins_mod.f90 instead of defining them here.
+BUILTIN_MAP_F90 = {
+    # Adding (scaled) fields
+    "X_plus_Y": DynXPlusYKern,
+    "inc_X_plus_Y": DynIncXPlusYKern,
+    "aX_plus_Y": DynAXPlusYKern,
+    "inc_aX_plus_Y": DynIncAXPlusYKern,
+    "inc_X_plus_bY": DynIncXPlusBYKern,
+    "aX_plus_bY": DynAXPlusBYKern,
+    "inc_aX_plus_bY": DynIncAXPlusBYKern,
+    # Subtracting (scaled) fields
+    "X_minus_Y": DynXMinusYKern,
+    "inc_X_minus_Y": DynIncXMinusYKern,
+    "aX_minus_Y": DynAXMinusYKern,
+    "X_minus_bY": DynXMinusBYKern,
+    "inc_X_minus_bY": DynIncXMinusBYKern,
+    # Multiplying (scaled) fields
+    "X_times_Y": DynXTimesYKern,
+    "inc_X_times_Y": DynIncXTimesYKern,
+    "inc_aX_times_Y": DynIncAXTimesYKern,
+    # Multiplying fields by a scalar (scaling fields)
+    "a_times_X": DynATimesXKern,
+    "inc_a_times_X": DynIncATimesXKern,
+    # Dividing (scaled) fields
+    "X_divideby_Y": DynXDividebyYKern,
+    "inc_X_divideby_Y": DynIncXDividebyYKern,
+    # Raising field to a scalar
+    "inc_X_powreal_a": DynIncXPowrealAKern,
+    # Setting field elements to scalar or other field's values
+    "setval_c": DynSetvalCKern,
+    "setval_X": DynSetvalXKern,
+    # Inner product of fields
+    "X_innerproduct_Y": DynXInnerproductYKern,
+    "X_innerproduct_X": DynXInnerproductXKern,
+    # Sum values of a field
+    "sum_X": DynSumXKern}
+
+
+# Built-in map dictionary in lowercase keys for invoke generation and
+# comparison purposes. This does not enforce case sensitivity to Fortran
+# built-in names.
+BUILTIN_MAP = get_builtin_map(BUILTIN_MAP_F90)
```

Comment:
I had a long, hard think about this. Am I correct in saying that the only purpose of BUILTIN_MAP_F90 is so that we can 'pretty print' the list of supported built-ins if we get one we don't recognise? I can't think of a better way of doing it but I do think we should rename things for clarity. e.g. "BUILTIN_MAP_F90" => "BUILTIN_MAP_CAPITALISED" and e.g. "get_builtin_map()" => "get_lowercase_builtin_map()".

Reply:
Yes, this is only for 'pretty print'. I made changes as requested. I also modified "builtin_map_f90_dict" to "builtin_map_capitalised_dict" to be consistent.

Reply:
BTW, does this file also need a change of copyright?

Reply:
Added full BSD license as agreed.